app/main.py

```diff
@@ -1,2 +1,62 @@
 class Dictionary:
-    pass
+    def __init__(self):
+        self.dict_len = 8
+        self.element = [None] * self.dict_len
+
+    def __setitem__(self, key, value):
+        if len(self) == int(self.dict_len * 2 / 3):
+            self.resize()
+
+        hash_key = hash(key)
+        new_el = (hash_key, key, value)  # store: hash, key, value
+        i = hash_key % self.dict_len
+
+        while True:
+            if self.element[i] is None:
+                self.element[i] = new_el
+                break
+            elif hash_key == self.element[i][0] and key == self.element[i][1]:
+                self.element[i] = new_el
+                break
+            else:
+                i = (i + 1) % self.dict_len
+
+    def __getitem__(self, key):
+        hash_key = hash(key)
+        i = hash_key % self.dict_len
+
+        if self.element[i] is None:
+            raise KeyError
+
+        while True:
+            if self.element[i][0] == hash_key:
+                return self.element[i][2]
+            else:
+                i = (i + 1) % self.dict_len
+
+    def __len__(self):
+        length = 0
+
+        for i in range(self.dict_len):
+            if self.element[i] is not None:
+                length += 1
+
+        return length
+
+    def resize(self):
+        new_dict_len = self.dict_len * 2
+        new_dict = [None] * new_dict_len
+
+        for i in range(self.dict_len):
```

Comment:
It's better to use the __setitem__ method to add items from the old list
It would help to avoid code duplication